src/main/webapp/app/instructor-course-dashboard/instructor-course-dashboard.component.ts

```diff
@@ -0,0 +1,113 @@
+import {JhiAlertService} from 'ng-jhipster';
+import {Component, OnInit} from '@angular/core';
+import {Course, CourseService, StatsForInstructorDashboard} from 'app/entities/course';
+import {ActivatedRoute} from '@angular/router';
+import {HttpErrorResponse, HttpResponse} from '@angular/common/http';
+import {InitializationState, Participation} from 'app/entities/participation';
+import {getIcon, getIconTooltip} from 'app/entities/exercise';
+import {Result, ResultService} from 'app/entities/result';
+import {TutorLeaderboardData} from 'app/instructor-course-dashboard/tutor-leaderboard/tutor-leaderboard.component';
+
+@Component({
+    selector: 'jhi-instructor-course-dashboard',
+    templateUrl: './instructor-course-dashboard.component.html',
+    providers: [JhiAlertService]
+})
+export class InstructorCourseDashboardComponent implements OnInit {
+    course: Course;
+
+    getIcon = getIcon;
+    getIconTooltip = getIconTooltip;
+
+    stats: StatsForInstructorDashboard;
+    dataNumbersForPieChart: number[];
+
+    tutorLeaderboardData: TutorLeaderboardData = {};
+
+    constructor(
+        private courseService: CourseService,
+        private resultService: ResultService,
+        private route: ActivatedRoute,
+        private jhiAlertService: JhiAlertService
+    ) {}
+
+    ngOnInit(): void {
+        this.loadCourse(Number(this.route.snapshot.paramMap.get('courseId')));
+    }
+
+    private loadCourse(courseId: number) {
+        this.courseService.findWithExercisesAndParticipations(courseId).subscribe(
+            (res: HttpResponse<Course>) => {
+                this.course = res.body;
+
+                let numberOfSubmissions = 0;
+                let numberOfAssessments = 0;
+
+                for (const exercise of this.course.exercises) {
+                    const validParticipations: Participation[] = exercise.participations.filter(participation => participation.initializationState === InitializationState.FINISHED);
+                    for (const participation of validParticipations) {
+                        for (const result of participation.results) {
```

Comment:
Maybe this could be done using the `.reduce((previous , current) => {}, initialValue)` function. However I am not sure if that makes it more readable.